permissions/helpers.py

```diff
@@ -0,0 +1,273 @@
+from django.contrib.auth.models import User
+from django.db.models import Q
+
+from applications.models import Application, ApplicationRound
+from reservation_units.models import ReservationUnit
+from reservations.models import Reservation
+from spaces.models import ServiceSector, Unit, UnitGroup
+
+
+def is_superuser(user: User):
+    return user.is_superuser
+
+
+def has_unit_group_permission(
+    user: User, unit_groups: list, required_permission: str
+) -> bool:
+    if not unit_groups:
+        return False
+
+    return user.unit_roles.filter(
+        unit_group__in=unit_groups,
+        role__permissions__permission=required_permission,
+    ).exists()
+
+
+def has_unit_permission(user: User, units: list, required_permission: str) -> bool:
+    if not units:
+        return False
+    unit_groups = []
+    for unit in units:
+        unit_groups += list(unit.unit_groups.all())
+
+    return user.unit_roles.filter(
+        Q(unit__in=units) | Q(unit_group__in=unit_groups),
+        role__permissions__permission=required_permission,
+    ).exists()
+
+
+def has_service_sector_permission(
+    user: User, service_sectors: list, required_permission: str
+) -> bool:
+    if not service_sectors:
+        return False
+    return user.service_sector_roles.filter(
+        service_sector__in=service_sectors,
+        role__permissions__permission=required_permission,
+    ).exists()
+
+
+def has_general_permission(user: User, required_permission: str) -> bool:
+    return user.general_roles.filter(
+        role__permissions__permission=required_permission
+    ).exists()
+
+
+def can_manage_general_roles(user: User) -> bool:
+    permission = "can_manage_general_roles"
+    return has_general_permission(user, permission) or is_superuser(user)
+
+
+def can_manage_service_sector_roles(user: User, service_sector: ServiceSector) -> bool:
```

Comment:
The might be some premature optimization in here with ordering, i'd guess is_superuser is the lightest of these to check, but as said more or less premature.

Reply:
I swapped lightest checks to beginning